app/controllers/sog/name-trainer.js

```diff
@@ -0,0 +1,145 @@
+import Controller from '@ember/controller';
+import { computed } from '@ember/object';
+import GroupMembershipsMixin from 'alpha-amber/mixins/group-memberships-mixin';
+import { run } from '@ember/runloop';
+
+export default Controller.extend(GroupMembershipsMixin, {
+  queryParams: ['group', 'difficulty'],
+  group: null,
+  difficulty: 1,
+  selectedGroup: null,
+  difficultyOptions: [
+    {
+      value: 1,
+      label: 'Makkelijk'
+    }, {
+      value: 2,
+      label: 'Gemiddeld'
+    }, {
+      value: 3,
+      label: 'Moeilijk'
+    }
+  ],
+  current: 1,
+  success: 0,
+  questions: [],
+  textInput: '',
+  started: false,
+  finished: false,
+  answered: false,
+  groupSelected: computed('group', 'model', function() {
+    this.generateQuestions(this.get('model'));
+    return this.get('group') != null;
+  }),
+  humanizedDifficulty: computed('difficultyOptions', 'difficulty', function() {
```

Comment:
difficultyOptions veranderd niet toch? Dan hoef je hem ook niet in de gaten te houden toch?

Reply:
Goeie, gaat er uit.